src/lib/message/L1ToL2Message.ts

```diff
@@ -245,17 +246,50 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     const creationReceipt =
       retryableCreationReceipt ||
       (await this.l2Provider.getTransactionReceipt(this.retryableCreationId))
-
-    if (creationReceipt) {
-      const l2Receipt = new L2TransactionReceipt(creationReceipt)
-      const redeemEvents = l2Receipt.getRedeemScheduledEvents()
+    if (creationReceipt){
+      const creationBlockNumber = creationReceipt.blockNumber
+      const ticketId = creationReceipt.transactionHash
+      const iFace = new Interface([
+        'event RedeemScheduled(     bytes32 indexed ticketId,     bytes32 indexed retryTxHash,     uint64 indexed sequenceNum,     uint64 donatedGas,     address gasDonor )',
+      ])
+      const redeemTopic = iFace.getEventTopic('RedeemScheduled')
+
+      let redeemEventLogs: ethers.providers.Log[] = []
+      let fromBlockNumber: number = creationBlockNumber
+      let fromBlock = await this.l2Provider.getBlock(fromBlockNumber)
+      let increment: number = 1000
+      const creationBlock = await this.l2Provider.getBlock(creationBlockNumber)
+      const maxBlock = await this.l2Provider.getBlockNumber()
+      while(fromBlockNumber <= maxBlock) {
+        const toBlockNumber = Math.min(fromBlockNumber + increment, maxBlock)
+        redeemEventLogs = await this.l2Provider.getLogs( {fromBlock: fromBlockNumber, toBlock: toBlockNumber, topics: [redeemTopic, ticketId]})
+        if (redeemEventLogs.length != 0) break
+        const toBlock = await this.l2Provider.getBlock(toBlockNumber)
+        if ((toBlock.timestamp - creationBlock.timestamp) > RETRYABLELIFETIMESECONDS) break
```

Comment:
we could theoretically get a false negative if a `keepalive` is used; not sure we need/want to handle this tho

Reply:
can add a optional flag that also look for the `LifetimeExtended` event to adjust the deadline accordingly, don't think we should make it the default behavior as it will double the `eth_getLog` calls

Reply:
I think we should look for the last redeem attempt and see if it was successful. If it wasnt then we can see if a keep alive occurred, and if it did continue to look for more redeems. I dont think we should have an optional flag though, since i dont think there are any occasions when a user would not want to know if a redeem happened in an extended period.

Reply:
Agree but how about we add the keepalive in another PR?